packages/ui/src/components/SelectInputV2/SelectBar.tsx

```diff
@@ -295,37 +337,105 @@ export const SelectBar = ({
+          index,
+        ) => {
+          const elementWidth = (currentValue as HTMLDivElement).offsetWidth
+          const newAccumulatedWidth =
+            accumulator.accumulatedWidth + elementWidth + SIZES_TAG.gap
+
+          return {
+            measuredVisibleTags: [
+              ...accumulator.measuredVisibleTags,
+              newAccumulatedWidth <= innerWidth &&
+                potentiallyNonOverflowedValues[index],
+            ].filter(Boolean) as OptionType[],
+            measuredHiddenTags:
+              accumulator.measuredHiddenTags +
+              (newAccumulatedWidth > innerWidth ? 1 : 0),
+            accumulatedWidth:
+              newAccumulatedWidth > innerWidth
+                ? accumulator.accumulatedWidth
+                : newAccumulatedWidth,
+            lastElementWidth:
+              index === toMeasureElementsArray.length - 1 ? elementWidth : 0,
+          }
+        },
+        {
+          measuredVisibleTags: [],
+          measuredHiddenTags: 0,
+          accumulatedWidth: 0,
+          lastElementWidth: 0,
+        },
+      )
+
+      const finalWidth =
+        accumulatedWidth +
+        (measuredHiddenTags ? SIZES_TAG.paddings + SIZES_TAG.plusTag : 0)
+      const overflowPx = finalWidth - innerWidth
+
+      // If only one element is selected and it is hidden, we need to show it
+      if (measuredHiddenTags === 1 && measuredVisibleTags.length === 0) {
+        setOverflowAmount(0)
+        setNonOverFlowedValues([potentiallyNonOverflowedValues[0]])
+
+        const newOverflowPx =
+          lastElementWidth +
+          (measuredHiddenTags - 1 > 0
+            ? SIZES_TAG.paddings + SIZES_TAG.plusTag
+            : 0) -
+          innerWidth
+        setLastElementMaxWidth(lastElementWidth - newOverflowPx)
+        setOverflow(true)
+      }
+
+      // If it overflows with the last tag, we need to add an ellipsis to the last element if there is enough space (>60px)
+      // else we hide it completely and add it to the overflow amount
+      else if (lastElementWidth - overflowPx > 60 && measuredHiddenTags > 0) {
+        setLastElementMaxWidth(lastElementWidth - overflowPx)
+        setOverflow(true)
+        setOverflowAmount(measuredHiddenTags)
+        setNonOverFlowedValues(measuredVisibleTags)
+      } else if (overflowPx > 0 && measuredHiddenTags > 0) {
+        setLastElementMaxWidth(0)
+        setOverflow(false)
+        setOverflowAmount(measuredHiddenTags + 1)
+        setNonOverFlowedValues(measuredVisibleTags.slice(0, -1))
+      }
+      // Otherwise, we have enough space to show all tags
+      else {
+        setOverflow(false)
+        setOverflowAmount(measuredHiddenTags)
+        setNonOverFlowedValues(measuredVisibleTags)
       }
     }
-    setNonOverFlowedValues(computedNonOverflowedValues)
-    setOverflowAmount(computedOverflowAmount)
-  }, [options, selectedData.selectedValues, width])
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Adding this comment means you are probable doing a useEffect on something you shouldn't which can create side effects.